### PR TITLE
Fixed tracing

### PR DESCRIPTION
**Pull request by leggebroten**
Fixed by cribbing from the open telemetry contributions.

https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex

### CODE REVIEW

File: lib/commanded/aggregate.ex [hunk 1]
```diff
@@ -47,25 +49,43 @@ defmodule OpentelemetryCommanded.Aggregate do
       "aggregate.lifespan": context.lifespan
     ]
 
-    Tracer.start_span("commanded:aggregate:execute", %{
-      kind: :consumer,
-      attributes: attributes
-    })
+    OpentelemetryTelemetry.start_telemetry_span(
+      @tracer_id,
+      "commanded.aggregate.execute",
```
Comment (leggebroten) [inline lib/commanded/aggregate.ex:54]: @davydog187 @bryannaegele 
Per your suggestion Bryan, the span names now use dots

File: lib/commanded/aggregate.ex [hunk 2]
```diff
@@ -47,25 +49,43 @@ defmodule OpentelemetryCommanded.Aggregate do
...
 
+    # try to normalize all errors to Elixir exceptions
     exception = Exception.normalize(kind, reason, stacktrace)
+
+    # record exception and mark the span as errored
     Span.record_exception(ctx, exception, stacktrace)
     Span.set_status(ctx, OpenTelemetry.status(:error, ""))
 
-    Tracer.end_span()
+    # do not close the span as endpoint stop will still be called with
```
Comment (davydog187) [inline lib/commanded/aggregate.ex]: was this copied from somewhere else? there's not "endpoint" in this context

File: lib/commanded/event_handler.ex [hunk 3]
```diff
@@ -52,23 +54,39 @@ defmodule OpentelemetryCommanded.EventHandler do
...
 
+    # try to normalize all errors to Elixir exceptions
     exception = Exception.normalize(kind, reason, stacktrace)
+
+    # record exception and mark the span as errored
     Span.record_exception(ctx, exception, stacktrace)
     Span.set_status(ctx, OpenTelemetry.status(:error, ""))
 
-    Tracer.end_span()
+    # do not close the span as endpoint stop will still be called with
```
Comment (davydog187) [inline lib/commanded/event_handler.ex]: same feedback

File: lib/commanded/aggregate.ex (hunk 2 above)
Comment (davydog187) [reply lib/commanded/aggregate.ex]: Have you validated this will result in the correct behavior? I use https://github.com/davydog187/foosball to test

**Comment by davydog187** [comment]
@leggebroten left you some small comments. Please take a look but this is looking great! 💪🏼

File: lib/commanded/aggregate.ex (hunk 2 above)
Comment (leggebroten) [reply lib/commanded/aggregate.ex]: Testing shows that this is wrong and do need to close the span

File: lib/commanded/event_handler.ex (hunk 3 above)
Comment (leggebroten) [reply lib/commanded/event_handler.ex]: Yup, same.  Testing showed the span needs to be closed here.

**Pull request merged**